mmdet3d/models/data_preprocessors/utils.py

```diff
@@ -63,3 +72,240 @@ def multiview_img_stack_batch(
         batch_tensor.append(
             F.pad(tensor, tuple(pad[idx].tolist()), value=pad_value))
     return torch.stack(batch_tensor)
+
+
+class _Voxelization(Function):
```

Comment:
move these new codes to another file in the same directory named voxelization.py